### PR TITLE
Added ExpandIcon widget.

### PR DESCRIPTION
**Pull request by dragostis**
Added a widget that automaitcally rotates the expansion icon and
offers a callback interface that captures presses.

### CODE REVIEW

File: packages/flutter/lib/src/material/expand_icon.dart [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'debug.dart';
+import 'icon.dart';
+import 'icons.dart';
+import 'icon_button.dart';
+import 'theme.dart';
+
+/// Callback triggered on expand/collapse state change. See [ExpandIcon].
```
Comment (Hixie) [inline packages/flutter/lib/src/material/expand_icon.dart]: Signature for...

File: packages/flutter/lib/src/material/expand_icon.dart [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+typedef void ExpandedCallback(bool isExpanded);
+
+/// A Material widget representing a rotating expand/collapse button. The icon
```
Comment (Hixie) [inline packages/flutter/lib/src/material/expand_icon.dart]: strike "Material"

File: packages/flutter/lib/src/material/expand_icon.dart [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+/// rotates 180 deg when pressed, then reverts the animation on a second press.
+/// The underlying icon is [Icons.expand_more].
+///
+/// See [IconButton].
+class ExpandIcon extends StatefulWidget {
+  /// Creates an [ExpandIcon] with the given padding, and a callback that is
+  /// triggered when the icon is pressed.
+  ExpandIcon({
+    Key key,
+    this.onPressed,
```
Comment (Hixie) [inline packages/flutter/lib/src/material/expand_icon.dart]: onPressed should be @required

File: packages/flutter/lib/src/material/expand_icon.dart [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+/// A Material widget representing a rotating expand/collapse button. The icon
+/// rotates 180 deg when pressed, then reverts the animation on a second press.
+/// The underlying icon is [Icons.expand_more].
+///
+/// See [IconButton].
+class ExpandIcon extends StatefulWidget {
+  /// Creates an [ExpandIcon] with the given padding, and a callback that is
+  /// triggered when the icon is pressed.
+  ExpandIcon({
```
Comment (Hixie) [inline packages/flutter/lib/src/material/expand_icon.dart]: this should take a size argument that sets the size of the icon

File: packages/flutter/lib/src/material/expand_icon.dart [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  bool _isExpanded = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = new AnimationController(duration: kThemeAnimationDuration);
+    _iconTurns = new Tween<double>(begin: 0.0, end: 0.5).animate(
+      new CurvedAnimation(
+        parent: _controller,
+        curve: Curves.ease
```
Comment (Hixie) [inline packages/flutter/lib/src/material/expand_icon.dart]: pretty sure this should be slowInFastOut

File: packages/flutter/lib/src/material/expand_icon.dart [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    if (config.onPressed != null)
+      config.onPressed(_isExpanded);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasMaterial(context));
+    return new IconButton(
+      padding: config.padding,
+      color: Colors.black38,
```
Comment (Hixie) [inline packages/flutter/lib/src/material/expand_icon.dart]: if onPressed is null, the button should be disabled

File: packages/flutter/test/material/expand_icon_test.dart [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+              expanded = !expanded;
+            }
+          )
+        )
+      )
+    );
+
+    expect(expanded, isFalse);
+    await tester.tap(find.byType(ExpandIcon));
+    expect(expanded, isTrue);
```
Comment (Hixie) [inline packages/flutter/test/material/expand_icon_test.dart]: test that it correctly handles being tapped half-way through the animation

Comment (Hixie) [inline packages/flutter/test/material/expand_icon_test.dart]: test that it handles being disabled

File: packages/flutter/lib/src/material/expand_icon.dart (hunk 6 above)
Comment (Hixie) [inline packages/flutter/lib/src/material/expand_icon.dart]: (meaning, looks disabled, doesn't react when tapped)

**Comment by Hixie** [comment]
LGTM, but in the future please focus on fixing existing PRs before adding more. You should ideally only be adding more PRs to the queue if your PRs already in the queue are awaiting review.

File: packages/flutter/lib/src/material/expand_icon.dart [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'debug.dart';
+import 'icon.dart';
+import 'icons.dart';
+import 'icon_button.dart';
+import 'theme.dart';
+
+/// Signature for expand/collapse state change. See [ExpandIcon].
```
Comment (Hixie) [inline packages/flutter/lib/src/material/expand_icon.dart]: come to think of it, maybe this should just be ValueChange<bool>

File: packages/flutter/lib/src/material/expand_icon.dart [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+typedef void ExpandedCallback(bool isExpanded);
+
+/// A widget representing a rotating expand/collapse button. The icon rotates
+/// 180 deg when pressed, then reverts the animation on a second press.
+/// The underlying icon is [Icons.expand_more].
+///
+/// See [IconButton].
```
Comment (Hixie) [inline packages/flutter/lib/src/material/expand_icon.dart]: You should say why IconButton may be interesting here

**Comment by HansMuller** [comment]
Can TwoLevelList incorporate this? That would prove its mettle :-)

**Pull request merged**